OrbitLinuxTracing/Utils.cpp

```diff
@@ -0,0 +1,153 @@
+#ifndef ORBIT_LINUX_TRACING_UTILS_H_
+#define ORBIT_LINUX_TRACING_UTILS_H_
+
+#include <fstream>
+#include <thread>
+
+#include "Logging.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/str_split.h"
+
+namespace LinuxTracing {
+
+std::optional<std::string> ReadFile(std::string_view filename) {
+  std::ifstream file(std::string{filename}, std::ios::in | std::ios::binary);
+  if (!file) {
+    ERROR("Could not open \"%s\"", std::string{filename}.c_str());
+    return std::optional<std::string>{};
+  }
+
+  std::ostringstream content;
+  content << file.rdbuf();
+  return content.str();
+}
+
+std::string ReadMaps(pid_t pid) {
+  std::string maps_filename = absl::StrFormat("/proc/%d/maps", pid);
+  std::optional<std::string> maps_content_opt = ReadFile(maps_filename);
+  if (maps_content_opt.has_value()) {
+    return maps_content_opt.value();
+  } else {
+    return "";
+  }
+}
+
+namespace {
```

Comment:
Please add a single anonymous namespace at the top of the file. It's harder to read when you are switching between anonymous and non-anonymous back and forth.

Reply:
I've heard this a few times, but take GetCgroupCpuset, GetCpusetCpusString, ParseCpusetCpus for example. They are all utility functions for GetCpusetCpus, putting them at the top far away from GetCpusetCpus makes no sense and is even worse for readability. Anonymous namespaces play more or less the role of the static keyword, but in that case there would be no need to put all static functions at the top.

Reply:
Yes, I understand why you are doing this. I disagree on what's more readable and easier to maintain, though. Keep rules on where to add stuff simple.

Reply:
So I thought about this and since I also agree with what you said that anonymous namespaces used like this are confusing because of the switching, I removed the anonymous namespace and used that static keyword, that for functions has the same effect.

Reply:
Sounds great!